packages/core/src/stepper/Step.tsx

```diff
@@ -0,0 +1,244 @@
+   * Optional string to determine the status of the step.
+   */
+  status?: StepStatus;
+  /**
+   * The stage of the step
+   */
+  stage?: StepStage;
+  /**
+   * Whether the step item is expanded.
+   */
+  expanded?: boolean;
+  /**
+   * Initial expanded state of the step.
+   */
+  defaultExpanded?: boolean;
+  /**
+   * Callback fired when the step is toggled.
+   */
+  onToggle?: ButtonProps["onClick"];
+}
+
+export type StepId = string;
+
+export type StepStatus = "warning" | "error";
+
+export type StepStage =
+  | "pending"
+  | "locked"
+  | "completed"
+  | "inprogress"
+  | "active";
+
+export type StepDepth = number;
+
+export interface StepProps
+  extends Omit<ComponentPropsWithoutRef<"li">, "onToggle"> {
+  label?: ReactNode;
+  description?: ReactNode;
+  status?: StepStatus;
+  stage?: StepStage;
+  expanded?: boolean;
+  defaultExpanded?: boolean;
+  onToggle?: ButtonProps["onClick"];
+}
+
+const withBaseName = makePrefixer("saltStep");
+
+export function Step({
+  id: idProp,
+  label,
+  description,
+  status,
+  stage = "pending",
+  expanded: expandedProp,
+  defaultExpanded,
+  onToggle,
+  className,
+  style,
+  children,
+  ...rest
+}: StepProps) {
+  const id = useId(idProp);
+  const targetWindow = useWindow();
+  const depth = useContext(StepDepthContext);
+  const orientation = useContext(StepperOrientationContext);
+
+  const hasNestedSteps = !!children;
+
+  const [expanded, setExpanded] = useControlled({
+    name: "Step",
+    state: "expanded",
+    controlled: expandedProp,
+    default: Boolean(defaultExpanded),
+  });
+
+  useComponentCssInjection({
+    testId: "salt-step",
+    css: stepCSS,
+    window: targetWindow,
+  });
+
+  useEffect(() => {
+    if (process.env.NODE_ENV !== "production") {
+      if (depth === -1) {
+        console.warn("<Step /> should be used within a <Stepper /> component!");
+      }
+      if (depth > 2) {
+        console.warn("<Step /> should not be nested more than 2 levels deep!");
+      }
+      if (orientation === "horizontal" && hasNestedSteps) {
+        console.warn(
+          "<Stepper /> does not support nested steps in horizontal orientation!",
+        );
+      }
+    }
+  }, [depth, orientation]);
+
+  const ariaCurrent = stage === "active" ? "step" : undefined;
+  const iconSizeMultiplier = depth === 0 ? 1.5 : 1;
+  const stageText = stage === "inprogress" ? "in progress" : stage;
+  const state = status || stageText;
+
+  const labelId = `${id}-label`;
+  const descriptionId = `${id}-description`;
+  const expandTriggerId = `${id}-expand-trigger`;
+  const nestedStepperId = `${id}-nested-stepper`;
+
+  const screenReaderOnly = {
+    stateId: `${id}-sr-only-state`,
+    stateText: state !== "active" ? state : "",
+    substepsId: `${id}-sr-only-substeps`,
+    substepsText: "substeps",
+    toggleSubstepsId: `${id}-sr-only-toggle-substeps`,
+    toggleSubstepsText: "toggle substeps",
+  };
+
+  return (
+    <li
+      id={id}
+      data-stage={stage}
```

Comment:
Not a blocker, but are these data attributes being used by anything within Salt, CSS or Cypress testing.